00-programming-fundamentals/git-04-merge-conflicts.md

```diff
@@ -0,0 +1,60 @@
+
+There are dedicated tools for manually resolving conflicts which are generally known as _three-way merge tools_. They are a bit like simplified text editors with specialized view layouts for displaying all three versions. Here's an example of merging a file with one such tool, [Beyond Compare](http://www.scootersoftware.com/):
+![Three-way merge tool](images/three-way-merge-tool.png)
+
+#### Resolving conflicts in Git
+Git's built-in approach for manually resolving conflicts does not require any specialized software, because it directly inserts _conflict markers_ into the conflicted file(s). By default however, it uses a two-way merge setup which shows only the **ours** and **theirs** versions. The simplest way to look at the **base** version is to go onto GitHub and look at a commit in the [repository's commit history](https://help.github.com/articles/differences-between-commit-views/).
+
+Here is the same file being merged, but with Git's conflict markers and the two versions it shows by default:
+
+```
+This is the content from
+<<<<<<< HEAD
+our version
+=======
+their version
+>>>>>>> c157db99220f47abfcb63547bfb8765ba571e19d
+```
+
+The conflict markers are the lines that start with `<<<<<<<`, `=======`, or `>>>>>>>`. All of the lines between the left-pointing brackets and the equals signs are the **ours** version, and all of the lines between the equals signs and the right-facing brackets are the **theirs** version.
+
+When manually resolving a merge conflict you need to figure out which version of each _conflict section_ should be used, or create a new version that mixes the two together. Determining which of these options to go with requires understanding the context of the code being changed, and the intention behind each of the changes. If you are unsure of how to resolve a conflict you should **discuss the situation with your teammate who made the other change**.
+
+Once you've determined what you need for the new, merged version of the code you should adjust the file to have exactly that code. **IMPORTANT**: You must also remove the conflict marker lines, as Git will not do so for you.
+
+#### If things go sideways
+It's possible that you'll encounter a merge conflict that your whole team is unsure about how to resolve. Maybe you're encountering a conflict against outdated code and doing `git pull` would make the conflict resolution much simpler. Or possibly in the process of resolving the conflict manually you've accidentally deleted a bunch of the file and can't undo it.
+
+In those cases Git provides you with a simple, effective escape hatch:
+```bash
+$ git merge --abort
+```
+
+This command will reset any changes made by Git or by you as part of a merge conflict. Everything will be put back exactly as it was before you ran the command that started the merge (generally either `git pull` or `git merge`).
```

Comment:
Can you add an "additional resources" section with one or two links to some sort of documentation on merge conflicts?